relay-config/src/config.rs

```diff
@@ -966,6 +971,9 @@ pub struct EnvelopeSpool {
     /// Defaults to 90% (5% less than max memory).
     #[serde(default = "spool_max_backpressure_memory_percent")]
     max_backpressure_memory_percent: f32,
+    /// Number of partitions of the buffer.
+    #[serde(default = "spool_envelopes_partitions")]
+    partitions: u32,
```

Comment:
Any particular reason for `u32` here? If you want to restrict the range, `NonZeroU8` might be an option.